04-cs-fundamentals/classroom/Arrays.md

```diff
@@ -1,55 +1,308 @@
+</details>
+
+<b>Exercise</b>: Modify the algorithm above to devise an algorithm that would return the minimum value in the input array. Try this algorithm and the one above on sample input values.
+
+### Sorted arrays
+There are several real world scenarios where we want our data to be ordered in ascending or descending way. Example:
+- Sort test scores in descending way
+- Sort song playtime in descending way
+- Sort identification numbers in ascending way
+- Sort names from *A* to *Z* (ascending way) like in the contacts' list on the phone.
+
+Having the data sorted, simplifies some of the algorithms.
+
+<b>Question</b>: If the array is sorted in ascending manner, how would you find the min value, or max value?
+<details>
+    <summary> Minimum and maximum value in a sorted array
+    </summary>
+        The algorithms for finding the minimum and maximum values get simplified if the array is sorted. The <b>minimum value</b> in an array that is sorted in an ascending manner will always be the first entry in the array i.e. array[0]. The <b>maximum value</b> in an array that is sorted in an ascending manner will always be the last entry in the array i.e. array[*n*-1], where *n* is the length of the array. If you compare the number of steps needed in this approach, with the approach needed for unsorted arrays, you'll find that the algorithm is significantly simpler and faster if the array is sorted.
+</details>
+
+You'll learn more about [algorithms to sort an array](Sorting.md) soon. For now, we can already see how having the array sorted makes finding the maximum and minimum value a faster and simplified algorithm. With that, let's consider searching for a value in a sorted array.
+
+### Binary Search
+The <b>binary search</b> algorithm leverages the sorted property while searching for a value in the array. Instead of comparing with each value in the array from index 0 to *n*-1, (where *n* is the number of elements in the array), we can be more methodic in eliminating where the value we are searching for will not be. We already apply the binary approach in our everyday life. E.g. Let's say we are looking for a word in dictionary, or while look for a name in the physical phone book. If the word/name starts with 'h', it will likely be in the first half of the book. We open the book to somewhere in the middle, compare the beginnig letter on that page to the word/name we are looking for. After comparing, we can safely eliminate the second half of the book. We keep eliminating half of the remaining pages in each step after repeating the process of opening a page to somewhere in the middle of valid pages and comparing. Unlike [linear search](#search), where after each comparison, one element is eliminated, with binary search, after each comparison, half of the remaining elements are eliminated.
```

Comment:
Just a note:  while HTML tags do work in github, in Markdown you technically make something bold by surrounding it with `**`.  Example:  `**this should be bold**`

Reply:
Ah! Let me make that formatting change in a separate update - since there are other files in the CS Fundamentals' curriculum that use the HTML tag for bolding.